src/server.ts

```diff
@@ -104,6 +111,28 @@ export async function start_server<RawConfiguration, Configuration, State>(
     }
   );
 
+  server.addHook("preHandler", (request, reply, done) => {
+    const expectedAuthHeader =
+      options.serviceTokenSecret === null
```

Comment:
Should we do a fuzzy check here rather than explicit null check?

Reply:
What happens for undefined?

Reply:
Correcting the TS types to reflect that the options are `undefined` if not set, rather than `null`. Then checking for undefined explicitly.

I don't want to do a fuzzy check since the secret could be set to an empty string.